src/couch_index/src/couch_index_updater.erl

```diff
@@ -208,12 +211,27 @@ update(Idx, Mod, IdxState) ->
 purge_index(Db, Mod, IdxState) ->
     {ok, DbPurgeSeq} = couch_db:get_purge_seq(Db),
     IdxPurgeSeq = Mod:get(purge_seq, IdxState),
-    if
-        DbPurgeSeq == IdxPurgeSeq ->
-            {ok, IdxState};
-        DbPurgeSeq == IdxPurgeSeq + 1 ->
-            {ok, PurgedIdRevs} = couch_db:get_last_purged(Db),
-            Mod:purge(Db, DbPurgeSeq, PurgedIdRevs, IdxState);
-        true ->
-            reset
+    if IdxPurgeSeq == DbPurgeSeq -> {ok, IdxState}; true ->
+        FoldFun = fun({PurgeSeq, _UUId, Id, Revs}, Acc) ->
+            Mod:purge(Db, PurgeSeq, [{Id, Revs}], Acc)
+        end,
```

Comment:
Understand that `couch_mrview_index:purge/4` or `couch_mrview_updater:purge/4` can deal with PurgedIdRevs (i.e. multiple Ids with Revs) in one call. However, when calling `couch_db:fold_purge_infos/5`, only one Id and Revs are available for one PurgeSeq in the callback.

Reply:
@jiangphcn We could always batch up in an accumulator and only invoke Mod:purge/4 occasionally rather than for individual rows.

@kocolosk The only reason its not batching was mulling over simple vs complex for questionable benefits. Given that purge is basically a "Break Glass in Case of Emergency" feature and shouldn't be part of an application's core workflow it seemed not overly useful to concern ourselves with performance.

Reply:
To clarify, not concerning ourselves with performance is more of a "No premature optimization" comment. We did of course make think a little so we didn't walk ourselves into some sort of `O(M^N)` exponential blow up thing. 😀

Reply:
Yeah, sure, I'm fine either way. I noticed that all the infrastructure seemed to be in place for it, and you were already doing some batching in internal replication (following the pattern that already existed).